web/packages/teleport/src/Login/useLogin.ts

```diff
@@ -39,7 +39,17 @@ export default function useLogin() {
   const auth2faType = cfg.getAuth2faType();
   const isLocalAuthEnabled = cfg.getLocalAuthFlag();
   const motd = cfg.getMotd();
-  const [showMotd, setShowMotd] = useState(!!motd);
+  const [showMotd, setShowMotd] = useState<boolean>(() => {
+    const redirectUri = history.getRedirectParam();
+
+    if (redirectUri?.includes('headless')) {
+      return false;
+    } else if (cfg.getMotd()) {
+      return true;
+    } else {
+      return false;
+    }
```

Comment:
```suggestion
    if (redirectUri?.includes('headless')) {
      return false;
    }
    return !!cfg.getMotd();
```

Reply:
this is much better. updated in https://github.com/gravitational/teleport/pull/31050/commits/4edc19f866d066fed7268b5d586f7c8e56592634